ext/json/json.c

```diff
@@ -159,7 +160,13 @@ PHP_JSON_API zend_result php_json_encode_ex(smart_str *buf, zval *val, int optio
 
 PHP_JSON_API zend_result php_json_encode(smart_str *buf, zval *val, int options) /* {{{ */
 {
-	return php_json_encode_ex(buf, val, options, JSON_G(encode_max_depth));
+	zend_string *indent_str = zend_string_init("    ", strlen("    "), 0);
```

Comment:
It [seems](https://github.com/php/php-src/search?q=zend_string_init) it could/should rather be:

```suggestion
	zend_string *indent_str = zend_string_init("    ", sizeof("    ") - 1, 0);
```

or [even](https://github.com/php/php-src/search?q=ZEND_STRL) better:

```suggestion
	zend_string *indent_str = zend_string_init(ZEND_STRL("    "), 0);
```

By the way, is it really OK to "hardcode" the length of 4 here regardless of `PHP_JSON_ENCODER_DEFAULT_INDENT` (especially looking how the existing code reads `JSON_G(encode_max_depth)` which is initialized with `PHP_JSON_PARSER_DEFAULT_DEPTH`)?

Reply:
Very good suggestion, didn't know about `ZEND_STRL` :).

Regarding the hardcoding of the indent for the internal api json encode, I think you're right. I have changed it so that it creates an indentation string based on the `PHP_JSON_ENCODER_DEFAULT_INDENT`.